sdk/ai/azopenai/client_extra.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+package azopenai
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+)
+
+type clientData struct {
+	endpoint string
+	azure    bool
+}
+
+func (client *Client) newError(resp *http.Response) error {
+	return newContentFilterResponseError(resp)
+}
+
+func newContentFilterResponseError(resp *http.Response) error {
+	//nolint this error is an azcore.ResponseError by definition.
+	respErr := runtime.NewResponseError(resp).(*azcore.ResponseError)
+
+	if respErr.ErrorCode != "content_filter" {
+		return respErr
+	}
+
+	body, err := runtime.Payload(resp)
+
+	if err != nil {
+		return err
+	}
+
+	var envelope *struct {
+		Error struct {
+			InnerError struct {
+				ContentFilterResults *ContentFilterResults `json:"content_filter_result"`
+			} `json:"innererror"`
+		}
+	}
+
+	if err := json.Unmarshal(body, &envelope); err != nil {
+		return err
+	}
+
+	return &ContentFilterResponseError{
+		ResponseError:        *respErr,
+		ContentFilterResults: envelope.Error.InnerError.ContentFilterResults,
+	}
+}
+
+func (client *Client) formatURL(path string, deployment *string) string {
```

Comment:
Similarly, we have custom_client.go. What's the distinction you're going for between client_extra and custom_client?

Reply:
(offline conversation - it's needed because of a chicken-and-egg situation with code gen)